### PR TITLE
Present status participants

### PR DESCRIPTION
**Pull request by Tomash-Klaus**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: EPlast/EPlast.DataAccess/Migrations/20220601164936_wasPresent_property_for_participant.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+
+        }
+
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+
+        }
```
Comment (tvardero) [inline EPlast/EPlast.DataAccess/Migrations/20220601164936_wasPresent_property_for_participant.cs:15]: Unnecessary migration

File: EPlast/EPlast.DataAccess/Migrations/20220601164936_wasPresent_property_for_participant.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace EPlast.DataAccess.Migrations
+{
+    public partial class wasPresent_property_for_participant : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+
```
Comment (tvardero) [inline EPlast/EPlast.DataAccess/Migrations/20220601164936_wasPresent_property_for_participant.cs:9]: Something not ok with this migration, column "WasPresent" is not generated.

File: EPlast/EPlast.DataAccess/Migrations/EPlastDBContextModelSnapshot.cs [hunk 3]
```diff
@@ -2023,6 +2023,9 @@ protected override void BuildModel(ModelBuilder modelBuilder)
                         .IsRequired()
                         .HasColumnType("nvarchar(450)");
 
+                    b.Property<bool>("WasPresent")
+                        .HasColumnType("bit");
```
Comment (tvardero) [inline EPlast/EPlast.DataAccess/Migrations/EPlastDBContextModelSnapshot.cs:2035]: No need, EF does it automatically

File: EPlast/EPlast.BLL/Services/Events/ParticipantManager.cs [hunk 4]
```diff
@@ -166,5 +166,16 @@ public async Task<IEnumerable<Participant>> GetParticipantsByUserIdAsync(string
...
+
+        public async Task<int> ChangeUserPresentStatus(int id)
+        {
+            var participant = await _repoWrapper.Participant
+                                   .GetFirstAsync(predicate: p => p.ID == id);
+            bool currentState = participant.WasPresent;
+            participant.WasPresent = !currentState;
+            _repoWrapper.Participant.Update(participant);
+            await _repoWrapper.SaveAsync();
+            return StatusCodes.Status204NoContent;
```
Comment (tvardero) [inline EPlast/EPlast.BLL/Services/Events/ParticipantManager.cs]: Return is not used

File: EPlast/EPlast.BLL/Interfaces/Events/IParticipantManager.cs [hunk 5]
```diff
@@ -68,7 +68,6 @@ public interface IParticipantManager
         /// </summary>
         /// <returns>Status code of the changing a present  status of the participant's event operation.</returns>  
         /// <param name="id">The Id of participant</param>
-        /// <response code="204">No content</response>
```
Comment (tvardero) [inline EPlast/EPlast.BLL/Interfaces/Events/IParticipantManager.cs]: Why?

File: EPlast/EPlast.WebApi/Controllers/EventsUsersController.cs [hunk 6]
```diff
@@ -48,14 +48,15 @@ private async Task<bool> HasAccessAsync()
...
+        //[HttpGet("{userId}")] 
         public async Task<IActionResult> GetEventUserByUserId(string userId)
         {
             var currentUserId = _userManager.GetUserId(User);
             if (currentUserId != userId && !(await HasAccessAsync()))
             {
                 return StatusCode(StatusCodes.Status403Forbidden);
             }
-
+            // �� ��� �������� 404 ���� � ���� �� ��������� �� ��������� ����� � ���� �� ��������� � ������
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Controllers/EventsUsersController.cs:59]: image

File: EPlast/EPlast.WebApi/Controllers/EventsUsersController.cs [hunk 7]
```diff
@@ -82,6 +83,8 @@ public async Task<IActionResult> GetEventsDataForCreate()
         /// <response code="201">Instance of EventCreateDTO</response>
         /// <response code="400">When the EventCreateDTO is null or empty</response> 
         [HttpPost("newEvent")]
+        //[HttpPost]
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Controllers/EventsUsersController.cs:86]: Delete this comment and empty line

File: EPlast/EPlast.WebApi/Controllers/EventsUsersController.cs [hunk 8]
```diff
@@ -120,10 +126,12 @@ public async Task<IActionResult> EventEdit(int eventId)
         /// <response code="204">Resource updated successfully</response>
         /// <response code="400">When the EventCreateDTO is null or empty</response>
         [HttpPut("editedEvent")]
+        //[HttpPut]
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Controllers/EventsUsersController.cs:129]: Delete comment

File: EPlast/EPlast.WebApi/Controllers/EventsUsersController.cs [hunk 9]
```diff
@@ -120,10 +126,12 @@ public async Task<IActionResult> EventEdit(int eventId)
...
         [Authorize(Roles = Roles.HeadsAndHeadDeputiesAndAdminAndPlastun)]
         public async Task<IActionResult> EventEdit([FromBody] EventCreateDTO createDTO)
         {
             await eventUserManager.EditEventAsync((createDTO));
+            // �� ������ ����� �������� ��� ������ 404 � 400
```
Comment (tvardero) [inline EPlast/EPlast.WebApi/Controllers/EventsUsersController.cs:134]: image

**Review by tvardero — CHANGES_REQUESTED** [review]
Read comments

File: EPlast/EPlast.BLL/Interfaces/Events/IParticipantManager.cs [hunk 10]
```diff
@@ -62,5 +62,11 @@ public interface IParticipantManager
         /// <returns>List of event participants.</returns>
         /// <param name="userId">The Id of logged in user</param>
         Task<IEnumerable<Participant>> GetParticipantsByUserIdAsync(string userId);
+
+        /// <summary>
+        /// Change present status of the participant's event.
+        /// </summary>
+        /// <param name="id">The Id of participant</param>
+        Task ChangeUserPresentStatus(int perticipantId);
```
[the remark this answers is not in the document]
Comment (tvardero) [reply EPlast/EPlast.BLL/Interfaces/Events/IParticipantManager.cs:70]: `ChangeUserPresentStatus` -> `ChangeUserPresentStatusAsync`

File: EPlast/EPlast.BLL/Services/Events/ParticipantManager.cs [hunk 11]
```diff
@@ -166,5 +166,15 @@ public async Task<IEnumerable<Participant>> GetParticipantsByUserIdAsync(string
 
             return participants;
         }
+
+        public async Task ChangeUserPresentStatus(int id)
+        {
+            var participant = await _repoWrapper.Participant
+                              .GetFirstAsync(predicate: p => p.ID == id);
```
Comment (tvardero) [inline EPlast/EPlast.BLL/Services/Events/ParticipantManager.cs:173]: GetFirstOrDefaultAsync()
if (part == null) throw KeyNotFoundException()

File: EPlast/EPlast.BLL/Interfaces/Events/IParticipantManager.cs (hunk 10 above)
[the remark this answers is not in the document]
Comment (Tomash-Klaus) [reply EPlast/EPlast.BLL/Interfaces/Events/IParticipantManager.cs:70]: changed all names

File: EPlast/EPlast.BLL/Services/Events/ParticipantManager.cs (hunk 11 above)
Comment (Tomash-Klaus) [reply EPlast/EPlast.BLL/Services/Events/ParticipantManager.cs]: done

File: EPlast/EPlast.WebApi/Controllers/EventsUsersController.cs [hunk 12]
```diff
@@ -136,6 +144,7 @@ public async Task<IActionResult> EventEdit([FromBody] EventCreateDTO createDTO)
         /// <response code="204">Resource updated successfully</response>
         /// <response code="400">When the Event is not approved</response>
         [HttpPut("approveEvent/{eventId}")]
+        //[HttpPatch("{eventId}")] new ????????
```
[the remark this answers is not in the document]
Comment (Tomash-Klaus) [reply EPlast/EPlast.WebApi/Controllers/EventsUsersController.cs]: done

File: EPlast/EPlast.WebApi/Controllers/EventsUsersController.cs (hunk 8 above)
Comment (Tomash-Klaus) [reply EPlast/EPlast.WebApi/Controllers/EventsUsersController.cs]: done

File: EPlast/EPlast.WebApi/Controllers/EventsUsersController.cs (hunk 7 above)
Comment (Tomash-Klaus) [reply EPlast/EPlast.WebApi/Controllers/EventsUsersController.cs]: done

**Review by tvardero — APPROVED** [review]

**Review by bkondrat — APPROVED** [review]

**Review by Omeliukh-Nazar — APPROVED** [review]

**Pull request merged**